core/src/com/unciv/ui/worldscreen/mainmenu/OptionsPopup.kt

```diff
@@ -329,6 +340,94 @@ class OptionsPopup(val previousScreen: BaseScreen) : Popup(previousScreen) {
+            mod.buildings,
+            mod.policies,
+            mod.nations,
+            mod.beliefs,
+        )
+
+        val filesToReplace = listOf(
+            "Units.json",
+            "TileImprovements.json",
+            "UnitPromotions.json",
+            "Buildings.json",
+            "Policies.json",
+            "Nations.json",
+            "Beliefs.json",
+        )
+
+        val deprecatedUniques = objectsToCheck
+            .flatMap { it.values }
+            .flatMap { it.uniqueObjects }
+            .filter { it.getDeprecationAnnotation() != null }
+
+
+        for (deprecatedUnique in deprecatedUniques) {
+            if (allDeprecatedUniques.contains(deprecatedUnique.text)) continue
+            allDeprecatedUniques.add(deprecatedUnique.text)
+
+            // note that this replacement does not contain conditionals attached to the original!
+            var uniqueReplacementText = deprecatedUnique.getReplacementText()
+            for (conditional in deprecatedUnique.conditionals)
+                uniqueReplacementText += " <${conditional.text}>"
+            val replacementUnique = Unique(uniqueReplacementText)
+
+            val modInvariantErrors = mod.checkUnique(
+                replacementUnique,
+                false,
+                "",
+                UniqueType.UniqueComplianceErrorSeverity.RulesetInvariant,
+                deprecatedUnique.sourceObjectType!!
+            )
+            for (error in modInvariantErrors)
+                println(error.text + " - " + error.errorSeverityToReport)
+            if (modInvariantErrors.isNotEmpty()) continue // errors means no autoreplace
+
+            if (mod.modOptions.isBaseRuleset) {
+                val modSpecificErrors = mod.checkUnique(
+                    replacementUnique,
+                    false,
+                    "",
+                    UniqueType.UniqueComplianceErrorSeverity.RulesetInvariant,
+                    deprecatedUnique.sourceObjectType
+                )
+                for (error in modSpecificErrors)
+                    println(error.text + " - " + error.errorSeverityToReport)
+                if (modSpecificErrors.isNotEmpty()) continue
+            }
+
+            deprecatedUniquesToReplacementText[deprecatedUnique.text] = uniqueReplacementText
+            println("Replace \"${deprecatedUnique.text}\" with \"$uniqueReplacementText\"")
+        }
+
+
+        val jsonFolder = mod.folderLocation!!.child("jsons")
+        for (fileName in filesToReplace) {
+            val file = jsonFolder.child(fileName)
+            if (!file.exists() || file.isDirectory) continue
+            var newFileText = file.readString()
+            for ((original, replacement) in deprecatedUniquesToReplacementText) {
+                newFileText = newFileText.replace("\"$original\"", "\"$replacement\"")
+            }
+            file.writeString(newFileText, false)
+        }
+        val toastText = "Replaced the following uniques:\n" +
```

Comment:
Due to the options itself being a popup, this toast doesn't show up until after closing the options screen